src/pipelines/rust_app.rs

```diff
@@ -291,26 +292,59 @@ pub struct RustAppOutput {
     pub wasm_output: String,
 }
 
+pub fn pattern_evaluate(pattern: String, params: &HashMap<String, String>) -> String {
+    let mut result = pattern.clone();
+    for (k, v) in params.iter() {
+        let needle = "{".to_owned() + k.as_str() + "}";
+        if k.starts_with("@") {
+            if let Ok(contents) = std::fs::read_to_string(v.as_str()) {
```

Comment:
```suggestion
        if v.starts_with("@") {
            if let Ok(contents) = std::fs::read_to_string(&v[1..]) {
```
Logically, it would seem that we should not drive this logic based on an overload of the key name. Instead, we should keep key names simple, and then if the value is prefixed with `@`, then that will be the indication that we need to read a file.

Even if the value only contains a single character `@`, this will still be safe: https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=7525e1389a63074c826a119d5007e877

Thoughts?

Reply:
Totally accepting your approach, putting @ in the key was lazy and probably less appropriate.